test/test_nn.py

```diff
@@ -333,13 +333,13 @@ def _forward_criterion(self, criterion, input, target):
             output = criterion(input, target)
         return output.item()
 
-    def _backward_criterion(self, criterion, input, target):
+    def _backward_criterion(self, criterion, input, target, gradOutput=torch.ones(())):
```

Comment:
Don't use mutable data structures as default args. The same value is used for every invocation.

Use `_backward_criterion(..., grad_output=None)` and then in the body:

```python
if grad_output is None:
   grad_output = torch.ones(())
````

Reply:
Yikes I didn't know that. Good catch!